mathesar_ui/src/components/cell-fabric/data-types/components/linked-record/LinkedRecordInput.svelte

```diff
@@ -51,23 +56,40 @@
     // If the value is cleared via a button, the focus may shift to that button.
     // We'd like to shift it back to the input element to that the user can
     // press `Enter` to launch the record selector.
-    element.focus();
+    element?.focus();
+  }
+
+  /**
+   * If this LinkedRecordInput in placed inside an AttachableDropdown, we want
+   * to tell the dropdown not to close when the user clicks within the Record
+   * Selector UI, so we set the modal element to "accompany" the dropdown.
+   */
+  function setRecordSelectorToAccompanyDropdown() {
+    const modal = document.querySelector<HTMLElement>('.modal-record-selector');
+    if (dropdownAccompanyingElements && modal) {
+      dropdownAccompanyingElements.add(modal);
+    }
```

Comment:
The modal element that's added here is never removed. While the modal itself is removed from the DOM, it's reference will be held here until the parent AttachableDropdown is removed.

While this should cause no major problems, I'm certain we'd have to do similar operations in other places and it would be better if we ensure that we always remove stale references.

Reply:
> it would be better if we ensure that we always remove stale references.

Fair enough. I added 4f017206804fe84e816821e685ac00dbca78d277 to address this.